_includes/seo.html

```diff
@@ -4,10 +4,15 @@
 {%- endif -%}
 {%- assign seo_url = seo_url | default: site.github.url -%}
 
-{% assign title_separator = site.title_separator | default: '-' | replace: '|', '&#124;' %}
+{% assign title_separator = site.title_separator | default: '-' %}
 
 {%- if page.title -%}
-  {%- assign seo_title = page.title | append: " " | append: title_separator | append: " " | append: site.title -%}
```

Comment:
I think the problem lies here: It assigns `page_title` (with fallback values) only when `page.title` is set, seems like circular logic to me.